identify/identify.py

```diff
@@ -141,6 +141,27 @@ def _shebang_split(line):
         return line.split()
 
 
+def _parse_nix_shebang(bytesio, cmd):
+    while bytesio.read(2) == b'#!':
+        next_line = bytesio.readline()
+        try:
+            next_line = next_line.decode('UTF-8')
+        except UnicodeDecodeError:
+            return cmd
+
+        for c in next_line:
+            if c not in printable:
+                return cmd
+
+        line_tokens = tuple(_shebang_split(next_line.strip()))
```

Comment:
Why did you revert to a potential fallback on line split?

Reply:
I misremembered how this used to work, and shlex.split can raise

I originally though this was a more true implementation of shebang parsing which would be `line.split(None, 1)` but it is not